Train.py

```diff
@@ -0,0 +1,448 @@
+        return train_info, test_info
+    
+    
+    def evaluate(self, config_path, model_path, optimizer_path, gpu_id):
+
+        """
+        Evaluate a model on test set.
+
+        Args:
+            config_path (str): Path to recover config dictionnary,
+            model_path (str): Path to recover model hyperparameters,
+            optimizer_path (str): Path to recover optimizer parameters.
+
+        Returns:
+            tuple: accuracy (float): Average accuracy on the test set,
+                   F1_score (float): Average F1 score on the test set.
+        """
+        
+        # Recover config files
+        with open(config_path) as f:
+            config = json.loads(f.read())
+            
+        # Recover model, optimizer and training configuration
+        training_config, model_config, optimizer_config = config['Training'], config['Model'], config['Optimizer']
+            
+        # Create dataloader
+        batch_size, num_workers = training_config['batch_size'], training_config['num_workers']
+        self.test_dataloader = get_dataloader(self.data_test, self.labels_test, batch_size, num_workers, balanced = False)
+        
+        # Load model
+        model = Transformer_classification(**model_config)
+        
+        # Move to gpu if available
+        available, device = define_device(gpu_id)
+        if available:
+            if torch.cuda.device_count() > 1:
+                model = torch.nn.DataParallel(model)
+        model.to(device)
+        
+        # Load model state
+        model.load_state_dict(torch.load(model_path))
+        
+        # Set evaluation mode
+        model.eval()
+        
+        # Load optimizer parameters
+        optimizer = torch.optim.Adam(model.parameters(), lr = optimizer_config['lr'],\
+                                          betas=(optimizer_config['b1'], optimizer_config['b2']))        
+        optimizer.load_state_dict(torch.load(optimizer_path))
+        
+        # Initialize accuracy
+        correct = 0
+        total = 0
+        f1_test = []
+        with torch.no_grad():
+            for data, labels in self.test_dataloader:
+                data = Variable(data.type(self.Tensor), requires_grad = True)
+                labels = Variable(labels.type(self.LongTensor))
+                data, labels = data.to(device), labels.to(device)
+                _, outputs = model(data)
+                predicted = torch.max(outputs.data, 1)[1]
+                total += labels.size(0)
+                correct += (predicted == labels).sum().item()
+                f1_test.append(f1_score(labels.cpu().detach().numpy(), predicted.cpu().detach().numpy(), average = 'macro'))
+
+        accuracy = correct / total 
+        F1_score = np.mean(f1_test)
+        
+        logger.info('Evaluation finished')
+        print("Accuracy on test set: {} \nF1 score on test set: {}".format(accuracy, F1_score))
+        
+        return accuracy , F1_score
+        
+
+def main(): 
```

Comment:
Add all of the possible combinations of inputs as examples here, so people know how to call your pipeline.
Define each input parameter (type, role)

Reply:
For instance example of config file, save_path,.. ?